### PR TITLE
Docblock updates

### PR DESCRIPTION
**Pull request by bryanagee**
In taking a look at the code, I found a lot of places where doc blocks would be helpful, so I'm adding those in. I'm putting the pull request in after adding just a few classes-worth, in hopes that I can get feedback and make sure this is both helpful and correct.

### CODE REVIEW

**Comment by matthewpatterson** [comment]
Hey @bryanagee, this looks good! I think I only have a couple points of feedback:

1. There are a lot of places with `@return array` where we do know the type of object in the array. For example, `Braintree_AddOn::all()` would return `Braintree_AddOn[]`.
2. There are some inconsistencies with newlines between `@param`, `@return`, and `@throws` annotations.

I'm not sure if you were planning to continue this for more of the classes. If not, I'm happy to clean these up when I merge. Thanks for doing this!

**Comment by bryanagee** [comment]
Thanks for the feedback. I was thinking that I'd continue just to get a handle on your code base, and possibly submit some cleanup after the docblocking is complete.

As for the array returns, I obviously didn't chase the method calls far enough to see how those types would resolve; it looks like the `Braintree_Util::extractAttributeAsArray()` actually uses a factory resolver to return an array of objects. A perfect example of why we need the docblocks =).

The one question remaining: is there a style guide that addresses docblock spacing and other issues? Specifically to the docblocking: it seems that PEAR and many older projects do not space the docblocking tags at all; newer ones vary a bit more. In the absence of a style guide, which should be applied?

**Comment by bryanagee** [comment]
Actually, in reviewing the code, the existing docblocks use no spacing between tags--regardless of type. I will go with that unless I hear otherwise. Thanks!

**Comment by bryanagee** [comment]
@matthewpatterson - do you want to review the changes I made to the prior submission and the newly added classes and see if that passes muster?

**Comment by bryanagee** [comment]
One important note--the version number deprecated tags was inferred from git blaming, and may not be accurate.

**Comment by bryanagee** [comment]
@matthewpatterson - Would you mind giving this a look before I continue? Just to make sure we're on the right road? Thanks!

**Comment by matthewpatterson** [comment]
@bryanagee Of course. Sorry I haven't had a lot of time to look at this yet, but I'll try to get to it today.

**Comment by bryanagee** [comment]
@matthewpatterson - no worries--I don't mean to rush you; just figured I would give it a bump while it's relatively fresh. If you don't get around to it, I'll just shelve it for a bit. Thanks!

File: lib/Braintree/CustomerGateway.php [hunk 1]
```diff
@@ -102,6 +102,7 @@ public function createNoValidate($attribs = array())
     /**
      * create a customer from a TransparentRedirect operation
      *
+     * @deprecated since version 2.35.0
```
Comment (matthewpatterson) [inline lib/Braintree/CustomerGateway.php]: Actually [2.3.0](https://github.com/braintree/braintree_php/commit/dfecae6e5e6dc45327d21e3be5dd1e21093648bf#diff-ba1450e872160266057a5277632b9969R113). Based on the change log for 2.3.0, I'd guess that all of the transparent redirect-related deprecations where in that version, but I'll confirm.

**Comment by matthewpatterson** [comment]
Like I mentioned inline, everything that you've marked as `@deprecated` was deprecated in 2.3.0. Otherwise, I think this looks great!

**Comment by bryanagee** [comment]
Awesome; I will update that version number and carry on. Thanks!

**Comment by matthewpatterson** [comment]
Hey @bryanagee, a couple people on our side are still giving this a look over, but would you be open to us merging these changes incrementally? After all your hard work, I don't want `master` to get so far ahead that this can't be merged.

**Comment by bryanagee** [comment]
Absolutely. Do you want commits to be one file (or a file and it's
respective gateway) at a time, and one branch per commit?

Bryan J. Agee
503.928.6799

**Comment by matthewpatterson** [comment]
I think one commit per branch is fine.

File: lib/Braintree/Gateway.php [hunk 2]
```diff
@@ -18,91 +22,163 @@ public function __construct($config)
         $this->config = $config;
     }
 
+    /**
+     * 
+     * @return \Braintree_AddOnGateway
```
Comment (chrissiedeist) [inline lib/Braintree/Gateway.php]: are the \'s meaningful? I'm guessing so because it's used in a number of the ones below, but didn't see it above

**Comment by chrissiedeist** [comment]
:+1:

**Comment by bryanagee** [comment]
The backslash shows that the classes referenced are in global namespace. It
doesn't make a difference right now, but makes it slightly more future
proof. Without them, adding a `namespace` block at the top of the file
would break the hinting unless the type was moved to the exact same
namespace as well.

Bryan J. Agee
503.928.6799

**Pull request merged**